src/pages/api/salesConfirmation.ts

```diff
@@ -59,30 +60,35 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
         }
 
         const uuid = getUuidFromCookie(req, res);
+        const schemeOp = isSchemeOperator(req, res);
 
         let userDataJson;
 
         const fareType = isFareType(fareTypeAttribute) && fareTypeAttribute.fareType;
 
-        if (fareType === 'single') {
-            userDataJson = getSingleTicketJson(req, res);
-        } else if (fareType === 'return') {
-            userDataJson = getReturnTicketJson(req, res);
-        } else if (fareType === 'period' || fareType === 'multiOperator') {
-            const ticketRepresentation = getSessionAttribute(req, TICKET_REPRESENTATION_ATTRIBUTE);
-            const ticketType = isTicketRepresentation(ticketRepresentation) ? ticketRepresentation.name : '';
-
-            if (ticketType !== 'geoZone' && ticketType !== 'multipleServices') {
-                throw new Error('No period type found to generate user data json.');
-            }
-
-            if (ticketType === 'geoZone') {
-                userDataJson = await getGeoZoneTicketJson(req, res);
-            } else if (ticketType === 'multipleServices') {
-                userDataJson = getMultipleServicesTicketJson(req, res);
+        if (schemeOp) {
+            userDataJson = await getSchemeOperatorTicketJson(req, res);
+        } else if (!schemeOp) {
```

Comment:
do we need the if(!schemeOp)?  as would be one or the other right?

Reply:
Good catch, I've changed this so that it is part of the existing else if chain